corehq/apps/users/models.py

```diff
@@ -1897,9 +1900,9 @@ def retire(self, deleted_by, deleted_via=None):
             pass
         else:
             django_user.delete()
-            if deleted_by:
-                log_model_change(deleted_by, django_user, message=f"deleted_via: {deleted_via}",
-                                 action=ModelAction.DELETE)
+        if deleted_by:
```

Comment:
is this optional just for unit tests? Should it validate it's only being called during testing? Similar comment for `unretire`

Reply:
That is correct.

When i did this in the PR that added this,
It was to avoid extra DB overhead for logging in tests and then not pass deleted_by in tests only.
Agree that updating this to check for `settings.UNIT_TESTING` would make sure it's only done in tests and should have been present already.